### ISSUE TITLE
Support D1 

### ISSUE THREAD

**Issue by brendonmatos** [cloudflare/terraform-provider-cloudflare#2544]
### Current Terraform and Cloudflare provider version

Terraform v1.4.6
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v4.7.1

### Description

Currently it is not possible to create a D1 instance via this provider.

### Use cases

https://developers.cloudflare.com/d1/

### Potential Terraform configuration

```hcl
resource "cloudflare_d1_database" "posts_db" {
  account_id = "xxxxxxxxxxx"
  name       = "posts"
}
```

**Comment by elithrar** [comment]
D1 will aim to have support by GA. Since D1 is still in alpha, we may make breaking changes (we have made several already) to the API surface as we iterate / act on feedback.

**Comment by xokaido** [comment]
How do we use this feature? The proposed code
```hcl
resource "cloudflare_d1_database" "posts_db" {
  account_id = "xxxxxxxxxxx"
  name       = "posts"
}
```
Doesn't work.

**Comment by jacobbednarz** [comment]
@xokaido are you running a dev build? if not, that is expected given it hasn't been released yet.

**Comment by xokaido** [comment]
@jacobbednarz 
I see the new version has been released ( 4.17 ) but we're getting an error:
![image](https://github.com/cloudflare/terraform-provider-cloudflare/assets/6562281/e78a346d-2cf9-41d6-b4f7-95d3e4f4ba5d)
I've tried multiple times, always the same response.
Can you help with that or whom do we address this issue?
